.gitlab-ci.yml

```diff
@@ -13,6 +13,12 @@ variables:
   RASPBIAN_URL: https://downloads.raspberrypi.org/raspios_lite_armhf/images/raspios_lite_armhf-2021-01-12/2021-01-11-raspios-buster-armhf-lite.zip
   RASPBIAN_NAME: 2021-01-11-raspios-buster-armhf-lite
 
+  # Which version of mender-convert to use in published image name. Normally
+  # empty, in which case the branch or tag name will be used, but this variable
+  # is available so that it's possible to make small build fixes on an already
+  # tagged version.
+  MENDER_CONVERT_PUBLISH_VERSION: ""
```

Comment:
I don't understand in which cases we are going to use this. Both for master and release branches the image will be continuously updated now, but I guess we will be using only tagged versions in the docs. Is it the idea that after a given release we can do some "hot fix" without tagging? Setting this to `2.5.0-1` or something like that?

If that is the case, I think is better to have this variable to hold only the `-1` and we concatenate it in the script.

Reply:
The main motivation is to be able to publish `<IMAGE>-mender-convert-2.5.0.img.xz`, IOW the one we are doing right now. We can't retag 2.5.0, but we know the only significant change is this `.gitlab-ci.yml` change, therefore it makes sense to publish it with a little cheat: It's not actually the pure 2.5.0 tag, but close enough, and manually verified by us.

The thinking is a little bit like how we publish images from dunfell. Even though we usually publish from the commit that becomes the tag, this is not enforced, and has certainly been sidestepped from time to time.

But I suppose it's possible that this is so rare that it doesn't make sense to submit this variable. We can just do it this once from a PR, and then close it without merging it afterwards.

Do you think we should remove it?

Reply:
You convinced my by aligning this idea to what we _can_ do with `meta-mender`. So let's keep it!

With great power comes great responsibility. We shall do these publishes with great care.